contracts/modules/Wallet/VestingEscrowWallet.sol

```diff
@@ -0,0 +1,525 @@
+    event RemoveTemplate(uint256 _index, uint256 _timestamp);
+
+    /**
+     * @notice Constructor
+     * @param _securityToken Address of the security token
+     * @param _polyAddress Address of the polytoken
+     */
+    constructor (address _securityToken, address _polyAddress)
+    public
+    Module(_securityToken, _polyAddress)
+    {
+    }
+
+    /**
+     * @notice Function used to initialize the different variables
+     * @param _treasury Address of the treasury
+     */
+    function configure(address _treasury, address _token) public onlyFactory {
+        require(_treasury != address(0), "Invalid address");
+        treasury = _treasury;
+        token = ERC20(_token);
+    }
+
+    /**
+     * @notice This function returns the signature of the configure function
+     */
+    function getInitFunction() public pure returns (bytes4) {
+        return bytes4(keccak256("configure(address,address)"));
+    }
+
+    /**
+     * @notice Used to deposit tokens from treasury
+     */
+    function depositTokens(uint256 _numberOfTokens) external withPerm(ADMIN) {
+        require(_numberOfTokens > 0, "Should be greater than zero");
+        token.safeTransferFrom(treasury, this, _numberOfTokens);
+        unassignedTokens = unassignedTokens.add(_numberOfTokens);
+        /*solium-disable-next-line security/no-block-members*/
+        emit DepositTokens(_numberOfTokens, now);
+    }
+
+    /**
+     * @notice Sends unassigned tokens to treasury
+     */
+    function sendToTreasury() external withPerm(ADMIN) {
+        uint256 amount = unassignedTokens;
+        unassignedTokens = 0;
+        token.safeTransfer(treasury, amount);
+        /*solium-disable-next-line security/no-block-members*/
+        emit SendToTreasury(amount, now);
+    }
+
+    /**
+     * @notice Sends available tokens to beneficiary
+     * @param _beneficiary beneficiary's address
+     */
+    function sendAvailableTokens(address _beneficiary) public withPerm(ADMIN) {
+        _sendTokens(_beneficiary);
+    }
+
+    /**
+     * @notice Used to withdraw available tokens by beneficiary
+     */
+    function withdrawAvailableTokens() external {
+        _sendTokens(msg.sender);
+    }
+
+    /**
+     * @notice Add template
+     * @param _numberOfTokens number of tokens
+     * @param _duration vesting duration
+     * @param _frequency vesting frequency
+     */
+    function addTemplate(uint256 _numberOfTokens, uint256 _duration, uint256 _frequency) external withPerm(ADMIN) {
+        _validateTemplate(_numberOfTokens, _duration, _frequency);
+        Template memory template;
+        template.numberOfTokens = _numberOfTokens;
+        template.duration = _duration;
+        template.frequency = _frequency;
+        templates.push(template);
+        /*solium-disable-next-line security/no-block-members*/
+        emit AddTemplate(_numberOfTokens, _duration, _frequency, now);
```

Comment:
emit the `index`  of the template if you are going to use the index further for any operation. no need to emit `now` as it automatically attaches with the events and we can read it from the blockchain until unless you are filtering according to timestamp